### PR TITLE
Fix Lobby forcing you from Favorites if you switched from the Serverlist before it was populated

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Fix a really annoying issue where if the serverlist takes a bit to load, and you only need to join via Favorites list, you're forced out of the Favorites tab into public Servers tab which can throw you off.

### CODE REVIEW

**Comment by Salanto** [comment]
Stop. Adding. Things. To. The. Milestone.

File: src/lobby.cpp [hunk 1]
```diff
@@ -524,6 +526,9 @@ void Lobby::list_servers()
 
 void Lobby::list_favorites()
 {
+  if (public_servers_selected) {
+    return;
+  }
```
Comment (Salanto) [inline src/lobby.cpp:531]:
```suggestion
```
This check is redundant as the only way to open the favorite menu is to use the UI element, meaning it will never auto-open, unlike the serverlist, which is opened when the client receives the serverlist from ms3.

File: src/lobby.cpp [hunk 2]
```diff
@@ -317,9 +317,9 @@ void Lobby::on_favorites_clicked()
 
   ao_app->set_favorite_list();
 
-  list_favorites();
-
   public_servers_selected = false;
+
+  list_favorites();
 }
```
Comment (Salanto) [inline src/lobby.cpp:323]: Unnecessary change, but eh, consistency I guess?

**Review by Salanto — CHANGES_REQUESTED** [review]
Not bad, not good either tho. 
Please take more time to figure out why it happens instead of layering a bandaid over it.

**Review by oldmud0 — APPROVED** [review]
Cool. This state machine logic has been ugly either way but it seems to make more sense now. I would consider consolidating `list_servers` and `list_favorites` but that would take some additional work.
